### PR TITLE
Sso sessions report / statistics enhancements

### PR DESCRIPTION
**Pull request by mmoayyed**
Closes https://github.com/Jasig/cas/issues/598

All credit goes to @mindblender

screen shot 2015-10-08 at 8 51 37 am
screen shot 2015-10-08 at 8 52 06 am

### CODE REVIEW

**Comment by mmoayyed** [comment]
There is still a bit of work to do, but I'll keep the pull open here as we are very close to the final point.

File: cas-server-webapp-support/src/main/java/org/jasig/cas/web/report/SingleSignOnSessionsReportController.java [hunk 1]
```diff
@@ -144,18 +188,119 @@ public boolean evaluate(final Object ticket) {
...
+     * @param type the type
+     * @return the sso sessions
+     */
+    @RequestMapping(value = "/getSsoSessions", method = RequestMethod.GET)
+    @ResponseBody
+    public Map<String, Object> getSsoSessions(@RequestParam(defaultValue = "ALL") final String type) {
+        final Map<String, Object> sessionsMap = new HashMap<>(1);
+        final SsoSessionReportOptions option = SsoSessionReportOptions.valueOf(type);
+
+        final Collection<Map<String, Object>> collection = getActiveSsoSessions(option);
```
Comment (leleuj) [inline cas-server-webapp-support/src/main/java/org/jasig/cas/web/report/SingleSignOnSessionsReportController.java]: I would not name `collection` the collection, but `activeSsoSessions`.

File: cas-server-webapp-support/src/main/java/org/jasig/cas/web/report/SingleSignOnSessionsReportController.java [hunk 2]
```diff
@@ -144,18 +188,119 @@ public boolean evaluate(final Object ticket) {
...
+        sessionsMap.put("activeSsoSessions", collection);
+
+        long totalTicketGrantingTickets = 0;
+        long totalProxyGrantingTickets = 0;
+        long totalUsageCount = 0;
+
+        final Set<String> uniquePrincipals = new HashSet<>();
+
+        for (final Map<String, Object> entry : collection) {
```
Comment (leleuj) [inline cas-server-webapp-support/src/main/java/org/jasig/cas/web/report/SingleSignOnSessionsReportController.java]: Nor `entry`, but `activeSsonSession`.

**Comment by leleuj** [comment]
Really nice UI improvement! +1 with some tiny comments...

**Comment by mmoayyed** [comment]
Updated the pull. Should be able to finalize by tomorrow.

**Comment by mmoayyed** [comment]
Updated docs

**Pull request merged**